assets/scripts/segments/view.js

```diff
@@ -201,6 +200,123 @@ function getGroundLevelOffset (elevation) {
+  //       // the sprite will be from the ground.
+  //       drawSegmentImage(sprite.id, ctx, undefined, undefined, width, undefined,
+  //         offsetLeft + ((repeatStartX + (i * (svg.width / TILE_SIZE_ACTUAL) * TILE_SIZE)) * multiplier),
+  //         (sprite.id.includes('ground') ? groundLevel : groundLevel - distanceFromGround),
+  //         width, undefined, multiplier, dpi)
+  //     }
+  //   }
+  // }
+
+  // if (graphics.left) {
+  //   const sprites = Array.isArray(graphics.left) ? graphics.left : [graphics.left]
+
+  //   for (let l = 0; l < sprites.length; l++) {
+  //     const sprite = getSpriteDef(sprites[l])
+  //     const svg = images.get(sprite.id)
+
+  //     const x = 0 + ((-left + (sprite.offsetX / TILE_SIZE_ACTUAL || 0)) * TILE_SIZE * multiplier)
+  //     const distanceFromGround = multiplier * TILE_SIZE * ((svg.height - (sprite.originY || 0)) / TILE_SIZE_ACTUAL)
+
+  //     drawSegmentImage(sprite.id, ctx, undefined, undefined, undefined, undefined,
+  //       offsetLeft + x,
+  //       groundLevel - distanceFromGround,
+  //       undefined, undefined, multiplier, dpi)
+  //   }
+  // }
+
+  // if (graphics.right) {
+  //   const sprites = Array.isArray(graphics.right) ? graphics.right : [graphics.right]
+
+  //   for (let l = 0; l < sprites.length; l++) {
+  //     const sprite = getSpriteDef(sprites[l])
+  //     const svg = images.get(sprite.id)
+
+  //     const x = (-left + actualWidth - (svg.width / TILE_SIZE_ACTUAL) - (sprite.offsetX / TILE_SIZE_ACTUAL || 0)) * TILE_SIZE * multiplier
+  //     const distanceFromGround = multiplier * TILE_SIZE * ((svg.height - (sprite.originY || 0)) / TILE_SIZE_ACTUAL)
+
+  //     drawSegmentImage(sprite.id, ctx, undefined, undefined, undefined, undefined,
+  //       offsetLeft + x,
+  //       groundLevel - distanceFromGround,
+  //       undefined, undefined, multiplier, dpi)
+  //   }
+  // }
+
+  if (graphics.center) {
+    const sprites = Array.isArray(graphics.center) ? graphics.center : [graphics.center]
+    console.log('sprites', { sprites })
+    const sprite = getSpriteDef(sprites[sprites.length - 1])
+    const svg = images.get(sprite.id)
+    const center = dimensions.center
+
+    const x = (center - ((svg.width / TILE_SIZE_ACTUAL) / 2) - left - (sprite.offsetX / TILE_SIZE_ACTUAL || 0)) * TILE_SIZE * multiplier
+    const distanceFromGround = multiplier * TILE_SIZE * ((svg.height - (sprite.originY || 0)) / TILE_SIZE_ACTUAL)
+
+    drawSegmentImage(sprite.id, ctx, undefined, undefined, undefined, undefined,
+      offsetLeft + x,
+      groundLevel - distanceFromGround,
+      undefined, undefined, multiplier, dpi)
+  }
+
+  if (type === 'sidewalk') {
+    drawProgrammaticPeople(ctx, segmentWidth / multiplier, offsetLeft - (left * TILE_SIZE * multiplier), groundLevel, randSeed, multiplier, variantString, dpi)
+  }
+}
+
```

Comment:
Is the only difference between these two functions the commented out part, or is there other changes?

This is a complex-enough function that duplicating it with very minor separate functionality gives me pause about accepting it this way.

Reply:
You're right, I've refactored this to use another param on the original method, let me know if you'd like to see it done differently!

Reply:
This looks much better, thanks!